bin_strat/indicators.py

```diff
@@ -0,0 +1,94 @@
+from enum import IntFlag
+import pandas as pd
+import numpy as np
+
+class TradingSignal(IntFlag):
+    BUY = 1
+    SELL = -1
+    HOLD = 0
+
+def macd_signal(price_history):
+    """
+    Calculates the most recent macd signal for a given asset
+    - Calculate the MACD and signal lines
+    - Generate the trading signal based off the lines crossing
+
+    For when we can do multiple orders/tickers:
+    - Screener itteration through multiple tickers to compare the Weighted Success Rate
+    - Checking the Binance API account balance for the current balance of the asset
+    Args:
+        price_history (pd.DataFrame): the historical price data for a given asset
+    Returns:
+        TradingSignal: the signal to sell (TradingSignal.SELL), buy (TradingSignal.BUY), or do nothing (TradingSignal.HOLD)
+    """
+    price_history_macd = _MACD_calc(price_history)
+    macd_signal = _MACD_strat(price_history_macd)
+    latest_signal = macd_signal['buy_sell_signal'].iloc[-1]
+
+    return latest_signal
+
+
+def _MACD_calc(price_history):
+    """
+    Calculates the MACD line and signal line
+    Args: price_history (pd.DataFrame): the historical price data for a given asset
+    Returns: price_history_macd (pd.DataFrame) with  columns ['Close', 'ret_pct_change', 'MACD', 'signal', 'go_long', 'potential_gains']
+    """
+    k = price_history['Close'].ewm(span=10, adjust=False).mean() # Get the 26-day EMA of the closing price
+    d = price_history['Close'].ewm(span=19, adjust=False).mean() # Get the 12-day EMA of the closing price
+    macd = k - d # Subtract the 26-day EMA from the 12-Day EMA to get the MACD
+    macd_s = macd.ewm(span=6, adjust=False).mean() # Get the 9-Day EMA of the MACD for the Trigger line
+    price_history_macd = price_history
```

Comment:
Careful with this assignment. In python, assigning an object to a variable copies a reference to the memory location, not the object itself. 

What that means in this example is that `price_history_macd` and `price_history` reference the same DataFrame. If you change `price_history`, you actually change `price_history_macd` as well. 

```
>>> df = pd.DataFrame({'open': [25, 90]}, index=['btc', 'eth'])
>>> df
     open
btc    25
eth    90
>>> df2 = df
>>> df2['close'] = [30, 95]
>>> df
     open  close
btc    25     30
eth    90     95
```

In modern programming you should very rarely (if ever) modify a function "parameter" because the code calling the function might not expect it. So in this case, the `price_history` is modified, and is also used later to calculate the buy price. Just something to consider.

Reply:
Oh ok, i thought i had to change it's name to reduce confusion of what the function inputs verse what it's output is going to be. I will fix. Does this reject the pull request?

Reply:
Sorry, meant to start a review instead of 1 comment, sec

Reply:
The naming is fine, it's the modifying that isn't. You could either copy the dataframe or do without one. It may seem inefficient to do that, but it's not that slow and saves you a headache later :*

Reply:
pushed fix